internal/sqladapter/testing/adapter.go.tpl

```diff
@@ -1078,6 +1078,91 @@ func TestDataTypes(t *testing.T) {
 	assert.Equal(t, testValues, item)
 }
 
+func TestBatchInsert(t *testing.T) {
+	sess := mustOpen()
+	defer sess.Close()
+
+	for batchSize := 0; batchSize < 17; batchSize++ {
+		err := sess.Collection("artist").Truncate()
+		assert.NoError(t, err)
+
+		batch := sess.InsertInto("artist").Columns("name").NewBatch(batchSize)
+
+		totalItems := int(rand.Int31n(21))
+
+		go func() {
+			for i := 0; i < totalItems; i++ {
+				batch.Values(fmt.Sprintf("artist-%d", i))
+			}
+			batch.Done()
+		}()
+
+		for q := range batch.Next() {
```

Comment:
How does `NextQuery()` sound? So this doesn't produce any confusion with `Iterator.Next()`, which returns a bool.